### ISSUE TITLE
Weird behavior in presence of template literal

### ISSUE THREAD

**Issue by qwemaze** [Glavin001/atom-beautify#1157]
Weird behavior in presence of template literal.

``` js
const composeUrl = ( host, api, key, data ) => {
        switch ( api ) {
            case 'Init':
                return `${host}/vwapi/Init?VWID=${key}&DATA=${encodeURIComponent(
            Object.keys(data).map((k) => `${k}=${data[k]}`).join(';')
        )}`;
        case 'Pay':
            return `${host}/vwapi/Pay?SessionId=${par}`;
    };
};
```

The beautified code should have looked like:

``` js
const composeUrl = ( host, api, key, data ) => {
    switch ( api ) {
        case 'Init':
            return `${ host }/vwapi/Init?VWID=${ key }&DATA=${ encodeURIComponent(
                Object.keys( data ).map( ( k ) => `${ k }=${ data[ k ] }` ).join( ';' )
            ) }`;
        case 'Pay':
            return `${ host}/vwapi/Pay?SessionId=${ par }`;
    };
};
```

Here is a link to the `debug.md` Gist: https://gist.github.com/anonymous/d9766518dc9dceb14ea07fa3ee5c9a11

**Comment by prettydiff** [comment]
Just as an FYI only the whitespace not contained by the escape delimiters `${` and `}` must be preserved.  The contents of the escape delimiters should be beautified as normal JavaScript.

**Comment by qwemaze** [comment]
There's actually two things:
- extra indentation fore some reason at ```switch``` (it appears to happen only with JS Beautify)
- not beautified contents of ```${ }``` (applies for JS Beautify and Prettydiff)

**Comment by prettydiff** [comment]
Pretty Diff appears to be beautifying code within the escape delimiters.  If you mean to have the extra spaces padding the `(` and `)` or `{` and `}` characters try Pretty Diff's *bracepadding* option.

**Comment by qwemaze** [comment]
Looks like PrettyDiff's spaces in paren not supported in atom-beautify
![](https://i.gyazo.com/0f59dc8bd916f8fd1e124a035e4b5232.png)

And if I use Pretty Diff as default beautifier it does this
```js
const composeUrl = (host, api, key, data) => {
    const path = `${host}/vwapi/${api}`;
    switch (api) {
        case 'Init':
            return `${path}?VWID=${key}&DATA=${encodeURIComponent(Object.keys(data).map((k) => `${k}=${data[k]}`).join(';'))}`;
        case 'Pay':
            return `${path}?SessionId=${data}`;
    };
};
```
All ```\n```'s and indentation inside template literals removed.

**Comment by prettydiff** [comment]
When I look at the original code sample I see two new lines in the template literal:

* The first before `Object.keys`
* The second after `.join(';')`

The first one is clearly within a template literal escape (immediately before `encodeURIComponent`) so there is no reason to preserve this white space as a part of the template literal unless there is a beautification rule about preserving as part of the escape.

The second one is in the same escape as the first one, but it is a bit more interesting because it follows a nested template literal.

**Comment by Glavin001** [comment]
https://github.com/Glavin001/atom-beautify/pull/1193 has been merged and published.

---

Published to v0.29.12
